php/class-customize-snapshot-manager.php

```diff
@@ -914,4 +915,56 @@ public function get_front_uuid_param() {
 	public function get_customize_uuid_param() {
 		return constant( get_class( $this->post_type ) . '::CUSTOMIZE_UUID_PARAM_NAME' );
 	}
+
+	/**
+	 * Let unauthenticated users see posts published in a changeset.
+	 *
+	 * @param array $allcaps All capabilities.
+	 * @param array $caps    Capabilities.
+	 * @param array $args    Args.
+	 * @return array All capabilities.
+	 */
+	public function filter_user_has_cap( $allcaps, $caps, $args ) {
+		global $current_user;
+
+		if (
+			! $this->current_snapshot_uuid
+			|| 0 !== $current_user->ID
```

Comment:
I think the check for current user being `0` should be removed because it should also work for under-privileged users making authenticated requests to read posts that normally they wouldn't be able to read, except that the post is made readable due to the `status` in the customized state.

Consider an author user making an authenticated API request. They should be able to access a